shells/browser/shared/src/main.js

```diff
@@ -45,37 +45,80 @@ function createPanelIfReactLoaded() {
       let render = null;
       let root = null;
 
+      // By having a bridgeId we ensure that we stop processing the old bridge right before the new bridge is created
+      // to prevent the old bridge from sending and receiving the messages meant for the newly created one.
+      let bridgeId = null;
+
       function initBridgeAndStore() {
-        let hasPortBeenDisconnected = false;
+        // Creating a bridge identifier separately from the `Bridge` instance and before that because
+        // the `listen` passed into `Bridge` constructor is called synchronously within this constructor,
+        // so we have no opportunity to assign the reference to the instance to a variable before this happens.
+        const thisBridgeId = (bridgeId = Math.random());
```

Comment:
Why not an autoincrementing counter? Might make it easier to log/debug too.

Reply:
Can be a counter. Will need a third variable to store the counter state. I tried it initially but it looked too much so I changed to random that holds this third variable implicitly.

Reply:
Just `let nextId = 0` and using `nextId++` seems fine to me.